signature/mechanism_openpgp.go

```diff
@@ -110,10 +110,16 @@ func (m *openpgpSigningMechanism) SupportsSigning() error {
 
 // Sign creates a (non-detached) signature of input using keyIdentity.
 // Fails with a SigningNotSupportedError if the mechanism does not support signing.
-func (m *openpgpSigningMechanism) Sign(input []byte, keyIdentity string) ([]byte, error) {
+func (m *openpgpSigningMechanism) SignWithOptions(input []byte, keyIdentity string, options *SignOptions) ([]byte, error) {
```

Comment:
It’s `SignWithPassphrase` now.

Please make `newGPGSigningMechanismInDirectory` and `newEphemeralGPGSigningMechanism` (in both implementations) return the private interface, that will prevent such mistakes in the future.